src/main/java/io/github/cottonmc/libcd/mixin/MixinAdvancementRewards.java

```diff
@@ -0,0 +1,77 @@
+package io.github.cottonmc.libcd.mixin;
+
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.github.cottonmc.libcd.api.AdvancementRewardsManager;
+import io.github.cottonmc.libcd.impl.CustomRewardsSettingsUtils;
+import net.minecraft.advancement.AdvancementRewards;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.util.Identifier;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.Map;
+
+@Mixin(AdvancementRewards.class)
+public class MixinAdvancementRewards implements CustomRewardsSettingsUtils {
+    private final Map<Identifier, Object> customRewardsSettings = Maps.newHashMap();
+
+    @Inject(method = "apply", at = @At("TAIL"))
+    public void onApply(ServerPlayerEntity serverPlayerEntity, CallbackInfo ci) {
+        AdvancementRewardsManager.APPLIERS.forEach((id, applier) ->
+                applier.accept(serverPlayerEntity, getRewardSettings(id))
+        );
+    }
+
+    @Inject(method = "toString", at = @At("TAIL"), cancellable = true)
+    public void onToString(CallbackInfoReturnable<String> cir) {
+        cir.setReturnValue(cir.getReturnValue().replace("}",
+                ", libcd:custom=[" + getAllRewardsSettings().toString() + "]}"
```

Comment:
`getAllRewardsSettings().toString()` is probably not gonna do what you want it to do, sadly. `HashMap` doesn't have a `toString` override, so it's just going to give you something like `HashMap@124BCFE1`.